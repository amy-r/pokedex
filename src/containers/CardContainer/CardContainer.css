.CardContainer {
  display: flex;
  align-items: center;
}