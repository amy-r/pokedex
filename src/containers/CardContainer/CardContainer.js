import React, { Component } from 'react';
import PropTypes, { shape, func, string } from 'prop-types';
import { connect } from 'react-redux';
import { fakeAction } from '../../actions';
import Card from '../../components/Card/Card';
import './CardContainer.css'
import Loading from '../../loading.gif';

export class CardContainer extends Component {

  render() {

  // if(this.props.loading) {
  //   return <img src = { Loading }/>
  // }

  const cards = this.props.pokeTypes.map( (obj) => {
    console.log(obj.name)
    return <Card name= {obj.name}/>
  })

    return (
      <div className = 'CardContainer'>

        {cards}
      </div>
    );
  }
}

CardContainer.propTypes = {
  pokeTypes: PropTypes.array
};

const mapStateToProps = ({ pokeTypes }) => ({ pokeTypes });


export default connect(mapStateToProps, null)(CardContainer);
