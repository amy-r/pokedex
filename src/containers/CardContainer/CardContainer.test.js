import React from 'react';
import ReactDOM from 'react-dom';
import {CardContainer} from './CardContainer';
import { shallow, mount } from 'enzyme';


describe('CardContainer', () => {
  let pokeTypes = [
    {id:'1',name:'normal',pokemon:['19','20']}, 
    {id:'2',name:'fighting',pokemon:['67','68']}
  ]

  it('matches the Snapshot', () => {
    const cardContainer = shallow(<CardContainer pokeTypes={pokeTypes}/>, { disableLifecycleMethods: true})
    expect(cardContainer).toMatchSnapshot();
  })

})
