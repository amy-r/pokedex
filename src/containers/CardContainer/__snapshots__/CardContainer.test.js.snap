// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    pokeTypes={
      Array [
        Object {
          "id": "1",
          "name": "normal",
          "pokemon": Array [
            "19",
            "20",
          ],
        },
        Object {
          "id": "2",
          "name": "fighting",
          "pokemon": Array [
            "67",
            "68",
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          name="normal"
        />,
        <Card
          name="fighting"
        />,
      ],
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "normal",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "fighting",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            name="normal"
          />,
          <Card
            name="fighting"
          />,
        ],
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "normal",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "fighting",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
